assets/sass/05-blocks/button/_editor.scss

```diff
@@ -1,36 +1,92 @@
 
+.wp-block-file .wp-block-file__button:active,
+.wp-block-search .wp-block-search__button:active {
+	color: var(--button--color-text-active);
+	background-color: var(--button--color-background-active);
+}
+
+.wp-block-file .wp-block-file__button:hover,
+.wp-block-search .wp-block-search__button:hover {
+	color: var(--button--color-text-hover);
+	background: transparent;
+}
+
 .wp-block-button__link {
 	// Extend button style
 	@extend %button-style;
+}
 
-	// Outline Style
-	&.is-style-outline,
-	.is-style-outline & {
-		color: var(--button--color-background);
-		background: transparent;
-		border: var(--button--border-width) solid currentColor;
+/**
+ * Block Options
+ */
+.wp-block-button {
 
-		&:visited {
-			color: var(--button--color-background);
-		}
+	&:not(.is-style-outline) {
+
+		.wp-block-button__link {
+
+			&:not(.has-background):not(.has-text-color) {
```

Comment:
```suggestion
			// Apply default button styles when custom text and background colors are not selected.
			&:not(.has-background):not(.has-text-color) {
```

Reply:
There might be a clearer way to word this. But in general I think it would be useful to add some comments about what the intention is.